core/service/json_parse.hpp

```diff
@@ -0,0 +1,77 @@
+#ifndef IROHA_JSON_PARSE_H
+#define IROHA_JSON_PARSE_H
+
+#include <string>
+#include <map>
+#include <memory>
+#include <vector>
+
+namespace json_parse {
+
+    enum class Type {
+        BOOL,
+        STR,
+        INT,
+        FLOAT,
+        LIST,
+        DICT
+    };
+
+    struct Object {
+        Type type;
+        int integer;
+        std::string str;
+        bool boolean;
+        float floating;
+
+        Object(Type t,int i) :
+            type(t), integer(i)
+        {
+            if(t != Type::INT) throw "TypeError"; //WIP
+        }
+
+        Object(Type t,std::string s) :
+            type(t), str(s)
+        {
+            if(t != Type::STR) throw "TypeError"; //WIP
+        }
+
+        Object(Type t,bool s) :
+            type(t), boolean(s)
+        {
+            if(t != Type::BOOL) throw "TypeError"; //WIP
+        }
+
+        Object(Type t,float f) :
+                type(t), floating(f)
+        {
+            if(t != Type::FLOAT) throw "TypeError"; //WIP
+        }
+
+
+        Object(Type t) :
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Struct 'Object' has a constructor with 1 argument that is not explicit.](https://www.codacy.com/app/soramitsu/iroha/file/4345391261/issues/source?bid=3712501&fileBranchId=3870324#l52)